internal/cmd/base/servers.go

```diff
@@ -548,20 +549,37 @@ func (b *Server) RunShutdownFuncs() error {
 }
 
 func (b *Server) ConnectToDatabase(dialect string) error {
-	dbase, err := gorm.Open(dialect, b.DatabaseUrl)
+	dbType, err := db.StringToDbType(dialect)
+	if err != nil {
+		return fmt.Errorf("unable to create db object with dialect %s: %w", dialect, err)
+	}
+	dbase, err := db.Open(dbType, b.DatabaseUrl)
 	if err != nil {
 		return fmt.Errorf("unable to create db object with dialect %s: %w", dialect, err)
 	}
 	if b.DatabaseMaxOpenConnections == 1 {
 		return fmt.Errorf("unable to create db object with dialect %s: %s", dialect, "max_open_connections must be unlimited by setting 0 or at least 2")
 	} else {
-		dbase.DB().SetMaxOpenConns(b.DatabaseMaxOpenConnections)
+		underlyingDB, err := dbase.DB()
+		if err != nil {
+			return fmt.Errorf("unable retreive db: %w", err)
+		}
+		underlyingDB.SetMaxOpenConns(b.DatabaseMaxOpenConnections)
```

Comment:
If, like my other comment suggests, we create a `db.DB` struct, then this functionality can move moved into the `db` package as a function on that struct, so this package doesn't need to access the underlying `sql.DB` directly, ie:

```go
package db

type DB {
  *gorm.DB
}

func (d *DB) SetMaxOpenConns(n int) error {
  underlyingDB, err := d.DB()
  if err != nil {
     return err
  }

  underlyingDB.SetMaxOpenConns(n)
  return nil
}
```

Reply:
this or something like it can be addressed in #1535